src/megaapi_impl.cpp

```diff
@@ -24446,12 +24424,11 @@ void MegaFolderDownloadController::downloadFolderNode(MegaNode *node, string *pa
 
     string localpath;
     client->fsaccess->path2local(path, &localpath);
-    FileAccess *da = client->fsaccess->newfileaccess();
+    auto da = client->fsaccess->newfileaccess();
     if (!da->fopen(&localpath, true, false))
     {
         if (!client->fsaccess->mkdirlocal(&localpath))
         {
-            delete da;
```

Comment:
Replace with da.reset()?

Reply:
in this case the fopen failed so it won't have an open handle - I've replaced it anyway since checkcomplete() will be called, just in case.